plugin/validator_test.go

```diff
@@ -2312,6 +2313,78 @@ func TestValidator_ValidateMethodsAndHeaders(t *testing.T) {
 	}
 }
 
+func TestValidator_NginxRegex(t *testing.T) {
+	schema, err := os.ReadFile("testdata/ngx-re-test.lua")
+	require.NoError(t, err)
+	require.NotNil(t, schema)
+
+	v, err := NewValidator(ValidatorOpts{})
+	require.NoError(t, err)
+	require.NotNil(t, v)
+	pluginName, err := v.LoadSchema(string(schema))
+	require.NoError(t, err)
+	require.Equal(t, "ngx-re-test", pluginName)
+
+	t.Run("overrides ngx.re.match and ngx.re.find", func(t *testing.T) {
+		schema, err := os.ReadFile("testdata/ngx-re-test.lua")
+		assert.Nil(t, err)
+		pluginName, err := v.LoadSchema(string(schema))
+		assert.EqualValues(t, "ngx-re-test", pluginName)
+		assert.Nil(t, err)
+
+		tests := []struct {
+			name        string
+			config      string
+			expectedErr string
+		}{
+			{
+				name: "ensure ngx.re.find is loaded and contains subject",
+				config: `"find_regex": "([0-9])",
+				"subject": "A123"`,
+			},
+			// {
+			// 	name: "ensure ngx.re.find works for validate_path",
+			// 	config: `"find_regex": "[[^[a-zA-Z0-9\.\-_~/%]*$]]",
+			// 	"subject": "/path"`,
+			// },
```

Comment:
I'm guessing what we want to do with these tests is validate some (all?) of the in-use regex strings. I would hope the existing tests that cover these functions would be sufficient, but I understand why we might want to directly test the behavior with specific unit tests.

Either way, when I placed the regex string found in [validate_path_with_regexes](https://github.com/Kong/goks/blob/main/lua-tree/share/lua/5.1/kong/db/schema/typedefs.lua#L431) under test, it is rejected with an unexpected error: `invalid character '.' in string escape code`. Dropping the escape characters from the regex resolves (`[[^[a-zA-Z0-9\.\-_~/%]*$]] -> [[^[a-zA-Z0-9.-_~/%]*$]]`). That would suggest that the existing regex isn't valid? Running the full test suite (the failing test excluded) does not result in any failures, so maybe this specific example isn't exercised by an existing test and I've found a regression.

I also noticed the following signature in use: `ngx.re.find("", path, "aj")`. It's interesting to me that the 3rd arg (`options`) [is a string originally](https://github.com/openresty/lua-nginx-module#ngxrefind), still a string [here](https://github.com/Kong/goks/blob/main/lua-tree/share/lua/5.1/kong/db/schema/typedefs.lua#L439)  but is rejected by our aliased gluare.re_find function as it requires an int, not a string.

My understanding was the overrides in `lualibs/go/ngx/loader.go` are global. These observations make it seem like that might not be so?